core/store/orm/orm.go

```diff
@@ -1488,6 +1488,46 @@ func toISO8601(t time.Time) string {
 		t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second(), t.Nanosecond(), tz)
 }
 
+const removeUnstartedTransactionsQuery = `
+WITH
+unstarted_job_runs AS (
+	DELETE FROM job_runs
+	WHERE status = 'unstarted'
+	RETURNING id, run_request_id
+),
+linked_run_requests AS (
+	DELETE FROM run_requests AS rr
+	WHERE rr.id IN (
+		SELECT run_request_id
+		FROM unstarted_job_runs
+	)
+),
+linked_task_runs AS (
```

Comment:
Are you absolutely sure this isnt already covered by foreign keys?

Please can you check this entire query for redundant deletes.

Reply:
I did check the cascading deletes, they execute like this:
```
(run_requests) --> (job_runs) --> (task_runs)
(eth_txes) --> (eth_tx_attempts) --> (eth_receipts)
```

So, the problem is that there is no cascading delete on the eth_txes side. If I delete a task run,  its associated eth_txes will not be deleted. 

I think that's OK, it makes sense. However, in order to remove associated eth_txes, I need to find all the ones linked with task_runs, before I delete the task_runs. Deleting job_runs does CASCADE to remove task_runs but there is no way of returning the IDs of the deleted task_runs so I can find the associated eth_txes. Another point is that all I have to start with here is the status of the job_run, ie. "unstarted". Any ideas for all this?

So, the first motivation for thiw query was to *control this deletion propagation sequence*. 
The second motivation for this query is *readability*. Cascading deletes are cool because they are automatic and atomic but they are also opaque: executing a delete can "cascade" into multiple tables with no feedback to the user. Finding out how deletes propagate is not easy either, see  https://stackoverflow.com/questions/37976832/how-to-list-tables-affected-by-cascading-delete.

Let me know what you think!

Reply:
This is about deleting unstarted transactions IIRC.

So what about forgetting the join onto task runs and simply `DELETE FROM eth_txes WHERE state = 'unstarted'`?